### PR TITLE
Routes - Replaced class by func. comp

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Type:** Refactor

**The following has been addressed in the PR:** #116

**Unit or Functional tests are included in the PR?**
 in progress

**Description:**

in order to use react hooks, in this PR I updated the routes by changing it from a class to a functional component. I have also splitted it's content so that it is clearer.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "extends": [
```
Comment (juanpicado) [inline tslint.json]: We do not use tslint anymore @priscilawebdev

Comment (priscilawebdev) [reply tslint.json]: I completely forgot, but then I remembered ... so I didn't finish

File: src/routers.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export enum Route {
+  ROOT = '/',
```
Comment (juanpicado) [inline src/routers.ts]: Let's keep this within the route file, they won't be used anywhere and not  change to be reused .

File: src/utils/package.ts [hunk 3]
```diff
@@ -57,10 +57,6 @@ export function formatDateDistance(lastUpdate): string {
   return distanceInWordsToNow(new Date(lastUpdate));
 }
 
-export function buildScopePackage(scope: string, packageName: string): string {
-  return `@${scope}/${packageName}`;
```
Comment (juanpicado) [inline src/utils/package.ts]: We need this method.

File: src/App/App.tsx [hunk 4]
```diff
@@ -85,6 +85,7 @@ export default class App extends Component<{}, AppStateInterface> {
   public loadOnHandler = async () => {
     try {
       const packages = await API.request<any[]>('packages', 'GET');
+      // @ts-ignore: FIX THIS TYPE:  Type 'any[]' is not assignable to type '[]'
```
Comment (juanpicado) [inline src/App/App.tsx]: I think `master` will solve this.

File: src/utils/package.ts (hunk 3 above)
Comment (priscilawebdev) [reply src/utils/package.ts]: Hmmm it's not being used anywhere else..only in the Version Page

File: src/App/App.tsx (hunk 4 above)
Comment (priscilawebdev) [reply src/App/App.tsx]: unfortunately it wasn't :/

File: src/App/App.tsx [hunk 5]
```diff
@@ -1,47 +1,23 @@
...
-  showLoginModal: boolean;
-  isLoading: boolean;
-}
-export default class App extends Component<{}, AppStateInterface> {
-  public state: AppStateInterface = {
+
+import AppRoute from './AppRoute';
+import { AppProps, AppContextProvider } from './AppContext';
+
+export default class App extends Component<{}, AppProps> {
```
Comment (tmkn) [inline src/App/App.tsx]: The `AppProps` name is a little misleading here. It's used for the state of a class component but its name is `AppProps` :/ It's not immediately clear that this is on purpose.
I would suggest renaming the import
```typescript
import { AppProps as AppState, AppContextProvider } from './AppContext';
```

Comment (priscilawebdev) [reply src/App/App.tsx]: Good point Tom, but as I am working on the refactor of the App Component, I believe we can merge like it is now 😉

Comment (juanpicado) [reply src/App/App.tsx]: Ok @priscilawebdev , it's fine, but I will memorize this :P later on.

**Pull request merged**